metpy/calc/tools.py

```diff
@@ -286,47 +287,134 @@ def reduce_point_density(points, radius, priority=None):
 
 
 @exporter.export
-def log_interp(x, xp, fp, **kwargs):
-    r"""Interpolates data with logarithmic x-scale.
+def log_interp(x, xp, *args, **kwargs):
+    r"""Interpolates data with logarithmic x-scale over a specified axis.
 
     Interpolation on a logarithmic x-scale for interpolation values in pressure coordintates.
 
     Parameters
     ----------
     x : array-like
-        The x-coordinates of the interpolated values.
+        1-D array of desired interpolated values.
 
     xp : array-like
         The x-coordinates of the data points.
 
-    fp : array-like
-        The y-coordinates of the data points, same length as xp.
+    args : array-like
+        The y-coordinates of the data points, same shape as xp.
+
+    axis : int, optional
+        The axis to interpolate over. Defaults to 0.
+
+    extrap: string, optional
+        Specify handling of interpolation points out of data bounds
+        'set_nan' : default, set value of points to 'nan'.
+        'none'    : return ValueError if data is out of bounds.
 
     Returns
     -------
     array-like
-        The interpolated values, same shape as x.
+        Interpolated values for each point with coordinates sorted in ascending order.
 
-    Examples
-    --------
-    >>> x_log = np.array([1e3, 1e4, 1e5, 1e6])
-    >>> y_log = np.log(x_log) * 2 + 3
-    >>> x_interp = np.array([5e3, 5e4, 5e5])
-    >>> metpy.calc.log_interp(x_interp, x_log, y_log)
-    array([ 20.03438638,  24.63955657,  29.24472675])
+    Notes
+    -----
+    xp and args must be the same shape.
 
     """
-    sort_args = np.argsort(xp)
+    # Pull out keyword args
+    extrap = kwargs.pop('extrap', 'set_nan')
+    axis = kwargs.pop('axis', 0)
 
+    # Remove units from x coords
```

Comment:
We could just have this function assume the same units and use a wrapper to take care of units for specific applications (like sigma to p).